src/core/unittest/TicketTest.cpp

```diff
@@ -103,6 +103,26 @@ struct TransportParamsScope {
     }
 };
 
+struct TicketScope {
+    uint8_t* p = nullptr;
+
+    ~TicketScope() {
+        reset();
+    }
+
+    uint8_t** operator&() {
+        reset();
```

Comment:
Why the reset here? I don't like this hidden side effect.

Reply:
I believe `wil::unique_ptr` uses the same contract, to prevent leaked memory. But I'll change it up.

Reply:
ComPtr uses a GetAddressOf() function to give this behavior and then just an operator uint8_t* to get the value without reset if necessary. That'd probably be what I'd do.

Reply:
Confirmed this is the way `wistd::unique_any_t` works, which is a base-class for `wistd::unique_ptr`: https://github.com/microsoft/wil/blob/ac455d2c2e3018dc37aafcdf0ffcc17adcd5739b/include/wil/resource.h#L319